liquibase: Add `sourceRoot`

alexanderheldt opened the pull request:
To fix the build issue that was introduced in f373abe60b11bf83c48a7fa84fc9f2c5c8d5151e
when the custom `unpackPhase` was removed without correctly validating the build,
which results in:
```
error: builder for '/nix/store/5qax6r6kxi4spn5bc7qjpgglxb9kvhxp-liquibase-4.29.2.drv' failed with exit code 1;
       last 3 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/v12lcsjxsq3n4lgi8za3k9mqc8pak59b-liquibase-4.29.2.tar.gz
       > unpacker produced multiple directories
       For full logs, run 'nix log /nix/store/5qax6r6kxi4spn5bc7qjpgglxb9kvhxp-liquibase-4.29.2.drv'.
```
---
I tested this by both installing the package as is, but also by overriding `version` and `src`:
```
(pkgs-unstable.liquibase.overrideAttrs (
  final: prev: {
    version = "4.18.0";
    src = pkgs.fetchurl {
      url = "https://github.com/liquibase/liquibase/releases/download/v${final.version}/liquibase-${final.version}.tar.gz";
      hash = "sha256-YRP2UtBqcVVtbtSouzcasthDAQywNlN56D34tFZKanY=";
    };
  }
))
```

---
Result of `nixpkgs-review pr 346148` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>liquibase</li>
  </ul>
</details>

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4636

SuperSandro2000 commented:
Can we set sourceRoot instead?

https://github.com/NixOS/nixpkgs/blob/066ee14086ae885ae29d28f77f0382fd7fbef0f9/pkgs/stdenv/generic/setup.sh#L1293

alexanderheldt committed:
liquibase: Add `sourceRoot`

To fix the build issue that was introduced in f373abe60b11bf83c48a7fa84fc9f2c5c8d5151e
when the custom `unpackPhase` was removed without correctly validating the build,
which results in:
```
error: builder for '/nix/store/5qax6r6kxi4spn5bc7qjpgglxb9kvhxp-liquibase-4.29.2.drv' failed with exit code 1;
       last 3 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/v12lcsjxsq3n4lgi8za3k9mqc8pak59b-liquibase-4.29.2.tar.gz
       > unpacker produced multiple directories
       For full logs, run 'nix log /nix/store/5qax6r6kxi4spn5bc7qjpgglxb9kvhxp-liquibase-4.29.2.drv'.
```

alexanderheldt commented:
> Can we set sourceRoot instead?
> 
> https://github.com/NixOS/nixpkgs/blob/066ee14086ae885ae29d28f77f0382fd7fbef0f9/pkgs/stdenv/generic/setup.sh#L1293

Yes, that seems like the nicer way to handle this. I force-pushed the change and updated the PR header + description.
